### ISSUE TITLE
Custom build (WSL, Ubuntu 18 or 20) can not be started: "Aborted (core dumped)"

### ISSUE THREAD

**Issue by janpio** [Expensify/Bedrock#844]
Follow up to https://github.com/Expensify/Bedrock/issues/843

```
sujan@Surface-Book-2:~/bedrock/Bedrock$ ./bedrock
Protip: check syslog for details, or run 'bedrock -?' for help
Aborted (core dumped)
```

I have no idea what I am doing, using `make` and all - but how can I debug this?

**Comment by coleaeason** [comment]
Can you please provide the bedrock logs out of `/var/log/syslog` for this crash?

**Comment by janpio** [comment]
After figuring out that this is not running by default on WSL and how to enable it with `sudo service rsyslog start` I can indeed to that:

```
sujan@Surface-Book-2:~/bedrock/Bedrock$ cat /var/log/syslog
Aug 14 21:49:38 Surface-Book-2 rsyslogd: set SO_TIMESTAMP failed on '/dev/log': Invalid argument [v8.32.0]
Aug 14 21:49:38 Surface-Book-2 rsyslogd: imklog: cannot open kernel log (/proc/kmsg): No such file or directory.
Aug 14 21:49:38 Surface-Book-2 rsyslogd: activation of module imklog failed [v8.32.0 try http://www.rsyslog.com/e/2145 ]Aug 14 21:49:38 Surface-Book-2 rsyslogd: rsyslogd's groupid changed to 106
Aug 14 21:49:38 Surface-Book-2 rsyslogd: rsyslogd's userid changed to 102
Aug 14 21:49:38 Surface-Book-2 rsyslogd:  [origin software="rsyslogd" swVersion="8.32.0" x-pid="1600" x-info="http://www.rsyslog.com"] start
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (main.cpp:314) main [main] [info] Starting bedrock server
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:1277) BedrockServer [main] [info] Loading plugins: CACHE, DB, JOBS, MYSQL
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (Cache.cpp:95) initCacheSize [main] [info] No -cache.max specified, defaulting to 16GB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (Cache.cpp:99) initCacheSize [main] [info] Initializing cache with maximum size of 17179869184 bytes
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:1303) BedrockServer [main] [info] Creating BedrockServer with plugins: Cache, DB, Jobs, MySQL
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:1339) BedrockServer [main] [info] Opening control port on 'localhost:9999'
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1639) S_socket [main] [info] DNS lookup took 1ms for 'localhost'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1651) S_socket [main] [info] Resolved localhost to ip: 127.0.0.1.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:1352) BedrockServer [main] [info] Launching sync thread 'sync'
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:1699) postPoll [main] [info] [performance] Read from 0 sockets, attempted to deserialize 0 commands, 0 were complete and deserialized in 0ms.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:185) sync [sync] [info] Setting dbPool size to: 250
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:71) SQLite [sync] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:115) SQLite [sync] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:332) _sqliteLogCallback [sync] [info] {SQLITE} Code: 283, Message: recovered 20 frames from WAL file /home/sujan/bedrock/Bedrock/bedrock.db-wal
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:139) SQLite [sync] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:152) SQLite [sync] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0000' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0001' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0002' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0003' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0004' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0005' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0006' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0007' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:2513) SQVerifyTable [sync] [info] 'journal0008' already exists, verifying.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1639) S_socket [sync] [info] DNS lookup took 0ms for 'localhost'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1651) S_socket [sync] [info] Resolved localhost to ip: 127.0.0.1.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLiteNode.cpp:64) SQLiteNode [sync] [info] {Surface-Book-2/SEARCHING} [NOTIFY] setting commit count to: 0
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLiteNode.cpp:1943) _changeState [sync] [info] {Surface-Book-2/SEARCHING} [NOTIFY] setting commit count to: 0
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:208) sync [sync] [info] Starting 8 worker threads.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:258) SQLite [blockingCommit] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:264) SQLite [blockingCommit] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:285) SQLite [blockingCommit] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:298) SQLite [blockingCommit] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLiteNode.cpp:504) update [sync] [hmmm] {Surface-Book-2/SEARCHING} No peers configured, jumping to LEADING
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLiteNode.cpp:1943) _changeState [sync] [info] {Surface-Book-2/SEARCHING} [NOTIFY] setting commit count to: 0
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLiteNode.cpp:1970) _changeState [sync] [info] {Surface-Book-2/SEARCHING} Switching from 'SEARCHING' to 'LEADING'
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:544) verifyTable [sync] [info] Creating 'cache': CREATE TABLE cache ( name TEXT NOT NULL PRIMARY KEY, value BLOB NOT NULL )
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLitePool.cpp:49) get [blockingCommit] [info] Returning new DB handle: 1
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:544) verifyTable [sync] [info] Creating 'cacheSize': CREATE TABLE cacheSize ( size INTEGER )
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:258) SQLite [worker1] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:264) SQLite [worker1] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:285) SQLite [worker1] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:298) SQLite [worker1] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:544) verifyTable [sync] [info] Creating 'jobs': CREATE TABLE jobs ( created TIMESTAMP NOT NULL, jobID INTEGER NOT NULL PRIMARY KEY, state TEXT NOT NULL, name TEXT NOT NULL, nextRun TIMESTAMP NOT NULL, lastRun TIMESTAMP, repeat TEXT NOT NULL, data TEXT NOT NULL, priority INTEGER NOT NULL DEFAULT 500, parentJobID INTEGER NOT NULL DEFAULT 0, retryAfter TEXT NOT NULL DEFAULT "")
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:568) verifyIndex [sync] [info] Verifying index 'jobsName'. isUnique? 0
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:578) verifyIndex [sync] [info] Creating index 'jobsName' on table 'jobs': ( name ). Executing 'CREATE INDEX jobsName ON jobs ( name )'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:568) verifyIndex [sync] [info] Verifying index 'jobsParentJobIDState'. isUnique? 0
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:578) verifyIndex [sync] [info] Creating index 'jobsParentJobIDState' on table 'jobs': ( parentJobID, state ) WHERE parentJobID != 0. Executing 'CREATE INDEX jobsParentJobIDState ON jobs ( parentJobID, state ) WHERE parentJobID != 0'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:568) verifyIndex [sync] [info] Verifying index 'jobsStatePriorityNextRunName'. isUnique? 0
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:578) verifyIndex [sync] [info] Creating index 'jobsStatePriorityNextRunName' on table 'jobs': ( state, priority, nextRun, name ). Executing 'CREATE INDEX jobsStatePriorityNextRunName ON jobs ( state, priority, nextRun, name )'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLitePool.cpp:49) get [worker1] [info] Returning new DB handle: 2
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:2138) _upgradeDB [sync] [info] Finished running DB upgrade.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:258) SQLite [worker2] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:264) SQLite [worker2] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:285) SQLite [worker2] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:298) SQLite [worker2] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLitePool.cpp:49) get [worker2] [info] Returning new DB handle: 3
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:258) SQLite [worker3] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:264) SQLite [worker3] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:285) SQLite [worker3] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:298) SQLite [worker3] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLitePool.cpp:49) get [worker3] [info] Returning new DB handle: 4
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:258) SQLite [worker4] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:264) SQLite [worker4] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:285) SQLite [worker4] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:298) SQLite [worker4] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLitePool.cpp:49) get [worker4] [info] Returning new DB handle: 5
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:258) SQLite [worker5] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:264) SQLite [worker5] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:285) SQLite [worker5] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:298) SQLite [worker5] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLitePool.cpp:49) get [worker5] [info] Returning new DB handle: 6
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:258) SQLite [worker6] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:264) SQLite [worker6] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:285) SQLite [worker6] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:298) SQLite [worker6] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLitePool.cpp:49) get [worker6] [info] Returning new DB handle: 7
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:258) SQLite [worker7] [info] Opening sqlite database: /home/sujan/bedrock/Bedrock/bedrock.db
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:264) SQLite [worker7] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Opening database '/home/sujan/bedrock/Bedrock/bedrock.db'.
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:285) SQLite [worker7] [info] {/home/sujan/bedrock/Bedrock/bedrock.db} Using SQLite default PRAGMA synchronous
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLite.cpp:298) SQLite [worker7] [info] Setting cache_size to 1048576KB
Aug 14 21:49:48 Surface-Book-2 bedrock: xxxxxx (SQLitePool.cpp:49) get [worker7] [info] Returning new DB handle: 8
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:1479) postPoll [main] [info] Ready to process commands, opening command port on 'localhost:8888'
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1639) S_socket [main] [info] DNS lookup took 0ms for 'localhost'.
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1651) S_socket [main] [info] Resolved localhost to ip: 127.0.0.1.
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (BedrockServer.cpp:1501) postPoll [main] [info] Opening port 'localhost:3306' for plugin 'MySQL'
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1639) S_socket [main] [info] DNS lookup took 0ms for 'localhost'.
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1651) S_socket [main] [info] Resolved localhost to ip: 127.0.0.1.
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:97) SException [main] [info] Throwing exception with message: 'couldn't bind' from libstuff/libstuff.cpp:1687
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (libstuff.cpp:1719) S_socket [main] [warn] Failed to open TCP port 'localhost:3306': couldn't bind(errno=13 'Permission denied')
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (STCPServer.cpp:21) openPort [main] [eror] Assertion failed: (port.s >= 0) != true
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (SLog.cpp:13) SLogStackTrace [main] [warn]
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (SLog.cpp:13) SLogStackTrace [main] [warn] SLogStackTrace() [0x7fd58ae419c9]
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (SLog.cpp:13) SLogStackTrace [main] [warn] STCPServer::openPort(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) [0x7fd58ae6cf17]
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (SLog.cpp:13) SLogStackTrace [main] [warn] BedrockServer::postPoll(std::map<int, pollfd, std::less<int>, std::allocator<std::pair<int const, pollfd> > >&, unsigned long&) [0x7fd58acec6b5]
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (SLog.cpp:13) SLogStackTrace [main] [warn] ./bedrock(main+0x3311) [0x7fd58acbe221]
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (SLog.cpp:13) SLogStackTrace [main] [warn] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fd5895c1b97]
Aug 14 21:49:49 Surface-Book-2 bedrock: xxxxxx (SLog.cpp:13) SLogStackTrace [main] [warn] ./bedrock(_start+0x2a) [0x7fd58acc091a]
```

And with `Failed to open TCP port 'localhost:3306': couldn't bind(errno=13 'Permission denied')` I could also figure out what was going on: I had a MySQL running locally on that port already. Killed it, and Bedrock starts 🚀
